src/styles/app-button/AppButton.tsx

```diff
@@ -0,0 +1,45 @@
+import { useButton } from "@react-aria/button";
+import { useRef } from "react";
+
+import classnames from "classnames";
+
+import "@/styles/app-button/AppButton.scss";
+import { AppButtonProps } from "@/styles/app-button/AppButton.types";
+
+const AppButton = function ({
+  variant = "primary",
+  size = "md",
+  full = false,
+  loading = false,
+  disabled = false,
+  inline = false,
+  children,
+  ...props
```

Comment:
I suggest also using `AsProps` here and allow button to behave as Link

Reply:
We can pass `elementType`:
https://react-spectrum.adobe.com/react-aria/useButton.html#usebutton

Reply:
Sounds great. Fixed